Gems/PhysX/Code/Editor/Source/ComponentModes/Joints/JointsSubComponentModeAngleCone.cpp

```diff
@@ -0,0 +1,432 @@
+                const float originalBaseY = tan(AZ::DegToRad(sharedState->m_startValues.first)) * Internal::ConeHeight;
+                const float newBaseY = originalBaseY + axisDisplacement;
+                const float newAngle = AZ::GetClamp(AZ::RadToDeg(atan(newBaseY / Internal::ConeHeight)), m_min, m_max);
+
+                EditorJointRequestBus::Event(
+                    m_entityComponentIdPair, &EditorJointRequests::SetLinearValuePair, m_propertyName,
+                    AngleLimitsFloatPair(newAngle, sharedState->m_startValues.second));
+
+                m_yLinearManipulator->SetBoundsDirty();
+            });
+
+        m_zLinearManipulator->InstallLeftMouseDownCallback(
+            [this, sharedState](const AzToolsFramework::LinearManipulator::Action& /*action*/) mutable
+            {
+                AngleLimitsFloatPair currentValue;
+                EditorJointRequestBus::EventResult(
+                    currentValue, m_entityComponentIdPair, &EditorJointRequests::GetLinearValuePair, m_propertyName);
+                sharedState->m_startValues = currentValue;
+            });
+
+        m_zLinearManipulator->InstallMouseMoveCallback(
+            [this, sharedState](const AzToolsFramework::LinearManipulator::Action& action)
+            {
+                AZ::Transform localTransform = AZ::Transform::CreateIdentity();
+                EditorJointRequestBus::EventResult(
+                    localTransform, m_entityComponentIdPair, &EditorJointRequests::GetTransformValue,
+                    JointsComponentModeCommon::ParamaterNames::Transform);
+                const AZ::Quaternion localRotation = localTransform.GetRotation();
+                const float axisDisplacement = action.LocalPositionOffset().Dot(localRotation.TransformVector(action.m_fixed.m_axis));
+                const float originalBaseZ = tan(AZ::DegToRad(sharedState->m_startValues.second)) * Internal::ConeHeight;
+                const float newBaseZ = originalBaseZ + axisDisplacement;
+                const float newAngle = AZ::GetClamp(AZ::RadToDeg(atan(newBaseZ / Internal::ConeHeight)), m_min, m_max);
+
+                EditorJointRequestBus::Event(
+                    m_entityComponentIdPair, &EditorJointRequests::SetLinearValuePair, m_propertyName,
+                    AngleLimitsFloatPair(sharedState->m_startValues.first, newAngle));
+
+                m_zLinearManipulator->SetBoundsDirty();
+            });
+
+        m_yzPlanarManipulator->InstallLeftMouseDownCallback(
+            [this, sharedState](const AzToolsFramework::PlanarManipulator::Action& /*action*/) mutable
+            {
+                AngleLimitsFloatPair currentValue;
+                EditorJointRequestBus::EventResult(
+                    currentValue, m_entityComponentIdPair, &EditorJointRequests::GetLinearValuePair, m_propertyName);
+                sharedState->m_startValues = currentValue;
+            });
+
+        m_yzPlanarManipulator->InstallMouseMoveCallback(
```

Comment:
question: I take it a lot of this code is just moved right?

Reply:
correct!